engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/JobRestServiceInteractionTest.java

```diff
@@ -186,7 +185,28 @@ public void testSetJobRetriesWithDueDate() {
   }
 
   @Test
-  public void testSetJobRetriesNonExistentJob() {
+  public void testSetRetriesWithNullDueDate() {
+    Map<String, Object> retriesVariableJson = new HashMap<>();
+    retriesVariableJson.put("retries", MockProvider.EXAMPLE_JOB_RETRIES);
+    retriesVariableJson.put("dueDate", null);
+
+    given()
+    .pathParam("id", MockProvider.EXAMPLE_JOB_ID)
+    .contentType(ContentType.JSON)
+    .body(retriesVariableJson).then().expect()
+    .statusCode(Status.NO_CONTENT.getStatusCode())
+    .when()
+    .put(JOB_RESOURCE_SET_RETRIES_URL);
+
+    verify(mockManagementService).setJobRetries(MockProvider.EXAMPLE_JOB_RETRIES);
+    verify(mockSetJobRetriesBuilder).jobId(MockProvider.EXAMPLE_JOB_ID);
+    verify(mockSetJobRetriesBuilder).dueDate(null);
+    verify(mockSetJobRetriesBuilder).execute();
+    verifyNoMoreInteractions(mockSetJobRetriesBuilder);
+  }
```

Comment:
🔧 Can we add a test where no due date is set?

Reply:
Maybe you can add the assertion `verify(mockSetJobRetriesBuilder, never()).dueDate(any());` to that test?

Reply:
As discussed, let's not add `verify(mockSetJobRetriesBuilder, never()).dueDate(any());` because `verifyNoMoreInteractions` covers cases where other unwanted methods are invoked.